sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -195,6 +195,41 @@ def test_glm_scalar_argument(Estimator, params, err_type, err_msg):
         glm.fit(X, y)
 
 
+@pytest.mark.parametrize(
+    "Estimator",
+    [GeneralizedLinearRegressor, PoissonRegressor, GammaRegressor, TweedieRegressor],
+)
+@pytest.mark.parametrize(
+    "params, err_type, err_msg",
+    [
+        (
+            {"alpha": [1, "2"]},
+            TypeError,
+            "alpha at index 1 must be an instance of float, not str.",
+        ),
+        (
+            {"alpha": [1, 2, 3], "fit_intercept": True},
+            ValueError,
+            "X width is 2 while alpha is of length 3",
+        ),
+        (
+            {"alpha": [1, 2, 3], "fit_intercept": False},
+            ValueError,
+            "X width is 2 while alpha is of length 3",
+        ),
+        ({"alpha": [-2, 2]}, ValueError, "alpha at index 0 == -2, must be >= 0.0"),
+        ({"alpha": [0.0, 0.0]}, ValueError, "X must be full rank when alpha = 0."),
+    ],
+)
+def test_glm_alpha_array(Estimator, params, err_type, err_msg):
+    """Test GLM for invalid alpha input when alpha is an iterable"""
+    X = [[1, 2], [2, 4]]
+    y = [1, 2]
+    glm = Estimator(**params)
+    with pytest.raises(err_type, match=err_msg):
+        glm.fit(X, y)
```

Comment:
do we already have a test that checks that these estimators do behave correctly when you pass an alpha as array. it seems you test the failure mode but you don't test that things work fine if you follow the API.

Reply:
I do have a proof of algorithmic correctness in my initial post. Does a test that checks that coefficients are equal if
`a == 0.1` vs `a == [0.1] * X.shape[1]` satisfactory?

Reply:
Another option is where I run a toy dataset using statsmodels, manually record down the coefficients and then run the corresponding regression using sklearn, then assert that the coefficients are equal to the ones produced in statsmodels (similar to my proof of correctness).